### PR TITLE
share core logic code with azure-springcloud-maven-plugin

### PR DESCRIPTION
**Pull request by wangmingliang-ms**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/springcloud/runner/deploy/SpringCloudDeploymentState.java [hunk 1]
```diff
@@ -175,142 +157,40 @@ protected String getDeployTarget() {
...
-                                            StringUtils.substring(fileName, 0, i),
-                                            StringUtils.substring(fileName, i + 1)
-                                    } :
-                                    new String[]{fileName, ""};
-                         })
-                         .filter(entry -> ArrayUtils.contains(springArtifacts, entry[0]))
-                         .collect(Collectors.toMap(entry -> entry[0], entry -> entry[1]));
+    @SneakyThrows
+    private static DeploymentResourceInner getInner(final SpringCloudDeploymentEntity deployment) {
+        final Field inner = SpringCloudDeploymentEntity.class.getDeclaredField("inner");
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/springcloud/runner/deploy/SpringCloudDeploymentState.java:188]: why we use reflection to get inner?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/springcloud/runner/deploy/SpringCloudDeploymentState.java [hunk 2]
```diff
@@ -175,142 +157,40 @@ protected String getDeployTarget() {
...
-    private void validateSpringCloudAppArtifact(String finalJar) throws AzureExecutionException, IOException {
-        final JarFile jarFile = new JarFile(finalJar);
-        final Attributes manifestAttributes = jarFile.getManifest().getMainAttributes();
-        final String mainClass = manifestAttributes.getValue(MAIN_CLASS);
-        if (StringUtils.isEmpty(mainClass)) {
-            throw new SpringCloudValidationException(String.format(MAIN_CLASS_NOT_FOUND, finalJar));
-        }
-        final String library = manifestAttributes.getValue(SPRING_BOOT_LIB);
-        if (StringUtils.isEmpty(library)) {
+    private void printPublicUrl(final SpringCloudApp app, @NotNull RunProcessHandler processHandler) {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/springcloud/runner/deploy/SpringCloudDeploymentState.java:163]: RunProcessHandler  should be replaced with more generate interface like IProgressReport

**Comment by andxu** [comment]
Update the dependency to fix the ci

File: PluginsAndFeatures/azure-toolkit-for-intellij/build.gradle [hunk 3]
```diff
@@ -102,6 +102,7 @@ dependencies {
     compile group: 'org.apache.commons', name: 'commons-text', version: '1.8'
     compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.10'
     compile 'net.minidev:json-smart:2.3'
+    compile 'com.microsoft.azure:azure-toolkit-springcloud-lib:0.0.1-SNAPSHOT', { force = true }
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/build.gradle:105]: 0.1.0

Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/build.gradle]: fixed.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/springcloud/runner/deploy/SpringCloudDeploymentState.java (hunk 1 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/springcloud/runner/deploy/SpringCloudDeploymentState.java:188]: add todo

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/springcloud/runner/deploy/SpringCloudDeploymentState.java (hunk 2 above)
Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/springcloud/runner/deploy/SpringCloudDeploymentState.java:163]: https://dev.azure.com/mseng/VSJava/_workitems/edit/1812811

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/springcloud/runner/deploy/SpringCloudDeploymentState.java (hunk 1 above)
Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/springcloud/runner/deploy/SpringCloudDeploymentState.java:188]: https://dev.azure.com/mseng/VSJava/_workitems/edit/1812809

**Review by andxu — APPROVED** [review]

**Pull request merged**